File: java/chain-of-responsibility/src/main/VicePresident.java
package main;

public class VicePresident extends ExpenseHandlerBase {
  @Override
  public void handleExpense(Expense expense) {
    if (expense.amount > 100 && expense.amount <= 1000) {
      System.out.println("Vice president handled the expense!");
    } else if (successor != null) {
      successor.handleExpense(expense);
    }
  }
}

File: csharp/ChainOfResponsibility/VicePresident.cs
﻿namespace ChainOfResponsibility;

public class VicePresident : ExpenseHandlerBase
{
    public override void HandleExpense(Expense expense)
    {
        if (expense.Amount > 100 && expense.Amount <=1000)
        {
            Console.WriteLine("Vice president handled the expense!");
        }
        else if (Successor != null)
        {
            Successor.HandleExpense(expense);
        }
    }
}